code/modules/SCPmodular/scps/euclid/SCP-173.dm

```diff
@@ -0,0 +1,212 @@
+	return FALSE
+
+/mob/living/simple_animal/hostile/scp_173/Initialize(mapload, var/mob/living/creator)
+	. = ..()
+	// Give spells
+	mob_spell_list += new /obj/effect/proc_holder/spell/aoe_turf/flicker_lights(src)
+	mob_spell_list += new /obj/effect/proc_holder/spell/aoe_turf/blindness(src)
+	mob_spell_list += new /obj/effect/proc_holder/spell/targeted/night_vision(src) //this is being kept just incase normal vision fucks up
+
+/mob/living/simple_animal/hostile/scp_173/proc/can_be_seen(turf/destination)
+	if(!cannot_be_seen)
+		return null
+	// Check for darkness
+	var/turf/T = get_turf(loc)
+	if(T && destination && T.lighting_object)
+		if(T.get_lumcount()<0.1 && destination.get_lumcount()<0.1) // No one can see us in the darkness, right?
+			return null
+		if(T == destination)
+			destination = null
+
+	// We aren't in darkness, loop for viewers.
+	var/list/check_list = list(src)
+	if(destination)
+		check_list += destination
+
+
+
+
+	// This loop will, at most, loop twice.
+	for(var/atom/check in check_list)
+		for(var/mob/living/M in viewers(world.view + 1, check) - src)
+			if(M.client && !M.has_unlimited_silicon_privilege)
+				if(!M.eye_blind)
+					if(next_blinks[M] == null)
+						next_blinks[M] = world.time+rand(15 SECONDS, 60 SECONDS)
+					return M
+		for(var/obj/mecha/M in view(world.view + 1, check)) //assuming if you can see them they can see you
+			if(M.occupant && M.occupant.client)
+				if(!M.occupant.eye_blind)
+					if(next_blinks[M.occupant] == null)
+						next_blinks[M.occupant] = world.time+rand(15 SECONDS, 60 SECONDS)
+					return M.occupant
+	return null
+
+
+
+
+
+
+/*    VIEWLOOP
+	// This loop will, at most, loop twice.
+	for(var/atom/check in check_list)
+		for(var/mob/living/M in viewers(world.view + 1, check) - src)
+			if(M.client && !M.has_unlimited_silicon_privilege)
+				if(!M.eye_blind)
+					return M
+		for(var/obj/mecha/M in view(world.view + 1, check)) //assuming if you can see them they can see you
+			if(M.occupant && M.occupant.client)
+				if(!M.occupant.eye_blind)
+					return M.occupant
+	return null
+*/
+
+
+/mob/living/simple_animal/hostile/scp_173/Move(turf/NewLoc)
+	if(can_be_seen(NewLoc))
+		if(client)
+			to_chat(src, "<span class='warning'>You cannot move, there are eyes on you!</span>")
+		return 0
+	return ..()
+
+/mob/living/simple_animal/hostile/scp_173/movement_delay()
+	return -5
+
+
+
+/mob/living/simple_animal/hostile/scp_173/Life()
+	. = ..()
+	if (isobj(loc))
+		return
+	var/list/our_view = view(src, 23)
+	for(var/A in next_blinks)
+		if(!(A in our_view))
+			next_blinks[A] = null
+			continue
+		if(world.time >= next_blinks[A])
+			var/mob/living/carbon/human/H = A
+			if(H.stat) // Sleeping or dead people can't blink!
+				next_blinks[A] = null
+				continue
+			H.visible_message("<span class='notice'>[H] blinks.</span>")
+			H.blind_eyes(2)
+			next_blinks[H] = 10+world.time+rand(15 SECONDS, 60 SECONDS)
```

Comment:
Jesus Christ.

Reply:
/spell/aoe_turf/blindness is broke
this was the workaround, people blink withing 23 tiles of 173
If theres a better way to do it please point me in the right direction